### PR TITLE
[Scenes] Legacy removal

### LINKED ISSUE #30256: [Scenes] Apply attributes fix once Zigbee legacy is removed in Spec
Once the fixes from https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/7910 are applied, the implementation needs to be modified accordingly.

### PR DESCRIPTION
**Pull request by lpbeliveau-silabs**
Applied changes suggested in https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/8668

Fixes: https://github.com/project-chip/connectedhomeip/issues/30256

This is skipping protocol compatibility because we are changing the cluster id and name as well in another PR.

### CODE REVIEW

**Comment by lpbeliveau-silabs** [comment]
This should not be merged before https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/8668 is merged in spec

**Review by bzbarsky-apple — COMMENTED** [review]
> This is skipping protocol compatibility because we are changing the cluster id and name as well in another PR.

The name change needs to happen either in this PR or first.  As written this will lead to broken APIs being shipped on Darwin....

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
Meant to request changes in https://github.com/project-chip/connectedhomeip/pull/31178#pullrequestreview-1795402272

**Review by jmartinez-silabs — APPROVED** [review]

File: src/app/clusters/scenes-server/SceneTable.h [hunk 1]
```diff
@@ -37,10 +37,9 @@ inline constexpr GroupId kGlobalGroupSceneId     = 0x0000;
 inline constexpr SceneIndex kUndefinedSceneIndex = 0xffff;
 inline constexpr SceneId kUndefinedSceneId       = 0xff;
 
-static constexpr size_t kIteratorsMax                 = CHIP_CONFIG_MAX_SCENES_CONCURRENT_ITERATORS;
-static constexpr size_t kSceneNameMaxLength           = CHIP_CONFIG_SCENES_CLUSTER_MAXIMUM_NAME_LENGTH;
-static constexpr size_t kScenesMaxTransitionTimeS     = 6000u;
-static constexpr size_t kScenesMaxTransitionTime100ms = kScenesMaxTransitionTimeS * 10;
+static constexpr size_t kIteratorsMax            = CHIP_CONFIG_MAX_SCENES_CONCURRENT_ITERATORS;
+static constexpr size_t kSceneNameMaxLength      = CHIP_CONFIG_SCENES_CLUSTER_MAXIMUM_NAME_LENGTH;
+static constexpr size_t kScenesMaxTransitionTime = 60000000u;
```
[the remark this answers is not in the document]
Comment (lpbeliveau-silabs) [reply src/app/clusters/scenes-server/SceneTable.h]: The test was added in the latest commit

File: src/app/tests/suites/certification/Test_TC_S_1_1.yaml
[the remark this answers is not in the document]
Comment (lpbeliveau-silabs) [reply src/app/tests/suites/certification/Test_TC_S_1_1.yaml:1]: YAML are currently out of date with the chip-test-plan new changes and will be updated in followup: https://github.com/project-chip/connectedhomeip/issues/31511#issue-2088365680

File: src/app/tests/suites/certification/Test_TC_S_2_2.yaml [hunk 2]
```diff
@@ -496,6 +496,34 @@ tests:
...
+      PICS: S.S.C00.Rsp && PICS_SDK_CI_ONLY
+      command: "AddScene"
+      arguments:
+          values:
+              - name: "GroupID"
+                value: G1
+              - name: "SceneID"
+                value: 0x01
+              - name: "TransitionTime"
+                value: 60000000
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_S_2_2.yaml:511]: I suggest a number just over 6000s.  For example, 7000000.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/tests/suites/certification/Test_TC_S_2_2.yaml (hunk 2 above)
Comment (lpbeliveau-silabs) [reply src/app/tests/suites/certification/Test_TC_S_2_2.yaml]: Applied

**Pull request merged**